src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1438,21 +1438,30 @@ void ShenandoahHeap::print_heap_regions_on(outputStream* st) const {
 
 size_t ShenandoahHeap::trash_humongous_region_at(ShenandoahHeapRegion* start) const {
   assert(start->is_humongous_start(), "reclaim regions starting with the first one");
-  assert(!start->has_live(), "liveness must be zero");
 
-  // Do not try to get the size of this humongous object. STW collections will
-  // have already unloaded classes, so an unmarked object may have a bad klass pointer.
-  ShenandoahHeapRegion* region = start;
-  size_t index = region->index();
-  do {
-    assert(region->is_humongous(), "Expect correct humongous start or continuation");
+  // Cannot access humongous_obj->size() in case class has been unloaded
```

Comment:
Why do we want to make two passes over the humongous object? I don't see why the original code needs to change or how this change is related to this PR?

Reply:
Thanks. We talked about this comment in the original PR: https://github.com/openjdk/jdk/pull/26256. It's a stale comment. I ran a pipeline with `gc*=trace` (and also inspected all usages).

Reply:
Running this change through test pipelines.  Perhaps the original problem that I had encountered was fixed in a different way upstream, and my change was no longer necessary or relevant after merging from upstream.